### ISSUE TITLE
How good is NTL with numerical category classification ?

### ISSUE THREAD

**Issue by KeshavSingh29** [tum-ai/number-token-loss#65]
Thanks for this amazing work and sharing the resources. 
While reading your paper, i had 2 questions: 
- How does NTL + CE loss in your case handle cases where numbers are just categories? For example in MCQ questions: the options could be 1,2,3,4 and there is essentially no regressive relation between the numbers and right choice here. Shouldnt NTL make this task even difficult ? 
- For tokenizer, you mention that digit level tokenizer work best with NTL. How does that work with decimal numbers ? If you can elaborate, that would be very helpful. 

Thanks!

**Comment by jannisborn** [comment]
Hi @KeshavSingh29,

Thanks for the interest in our work!

1. There is no special treatment in case a number happens to be a category. A token is a token and it's always processed the same way, no matter in what context it occurs. This is a fundamental principle in NLP that should not be touched. If we would start to apply different processing when a token like "bank" occurs in the context of finance vs furniture, we start to go back to traditional computational linguistics. The case you mention is an edge case and will not dominate the model's understanding of numbers. 
2. That's a good point. With digit-level tokenization a decimal number like `12.34` will be tokenized into `1 2 . 3` and `4`. In that case, NTL will be calculated on tokens `1 2 3` and `4`. Cross Entropy will still be calculated on all tokens. If you happen to have a multi digit tokenizer that tokenizes it into `12` `.` and `34`, then NTL will be calculated on tokens `12` and `34`

Let us know if you need more clarification!

**Comment by KeshavSingh29** [comment]
Thanks for the explaination @jannisborn ! That makes sense. 
Just to follow up briefly, 

- Does choosing different loss during pre-training (NTL+CE) to only CE during SFT make any difference ? Or is it something you dont recommend?
